django/db/models/base.py

```diff
@@ -1111,6 +1126,12 @@ def delete(self, using=None, keep_parents=False):
 
     delete.alters_data = True
 
+    async def adelete(self, using=None, keep_parents=False):
+        return await sync_to_async(self.delete)(
+            using=using,
+            keep_parents=keep_parents,
+        )
```

Comment:
As far as I'm aware, `adelete()` and `asave()` should have `alters_data` set to `True`.

Reply:
``alters_data`` is already True for ``delete`` and  ``save``, do i need to do it again for ``asave`` and ``adelete``?

Reply:
> `alters_data` is already True for `delete` and `save`, do i need to do it again for `asave` and `adelete`?

Yes, because the template system is not aware what methods are called inside.